src/pkg/packager/publish.go

```diff
@@ -0,0 +1,369 @@
+package packager
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/defenseunicorns/zarf/src/pkg/message"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+
+	"github.com/docker/cli/cli/config"
+	"github.com/docker/cli/cli/config/configfile"
+	v1name "github.com/google/go-containerregistry/pkg/name"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"oras.land/oras-go/v2"
+	"oras.land/oras-go/v2/content"
+	"oras.land/oras-go/v2/content/file"
+	"oras.land/oras-go/v2/registry/remote"
+	"oras.land/oras-go/v2/registry/remote/auth"
+	"oras.land/oras-go/v2/registry/remote/errcode"
+)
+
+// https://github.com/docker/docs/issues/8230
+const ociLayerLimit = 127
+
+// Publish publishes the package to a registry
+//
+// This is a wrapper around the oras library
+// and much of the code was adapted from the oras CLI - https://github.com/oras-project/oras/blob/main/cmd/oras/push.go
+//
+// Authentication is handled via the Docker config file created w/ `docker login`
+func (p *Packager) Publish() error {
+	p.cfg.DeployOpts.PackagePath = p.cfg.PublishOpts.PackagePath
+	if err := p.loadZarfPkg(); err != nil {
+		return fmt.Errorf("unable to load the package: %w", err)
+	}
+	spinner := message.NewProgressSpinner("")
+	defer spinner.Stop()
+
+	if p.cfg.PublishOpts.RegistryURL == "docker.io" {
+		// docker.io is commonly used, but is not a valid registry URL
+		p.cfg.PublishOpts.RegistryURL = "registry-1.docker.io"
+	}
+
+	// push everything
+	paths := []string{
+		filepath.Join(p.tmp.Base, "checksums.txt"),
+		filepath.Join(p.tmp.Base, "zarf.yaml"),
+		filepath.Join(p.tmp.Base, "sboms.tar.zst"),
+	}
+	componentTarballs, err := filepath.Glob(filepath.Join(p.tmp.Base, "components", "*.tar.zst"))
+	if err != nil {
+		return err
+	}
+	paths = append(paths, componentTarballs...)
+	if p.cfg.PublishOpts.IncludeImages {
```

Comment:
I would rather handle this in a smarter way than including the images and failing if there are too many layers.  We could either include it as another artifact (tagging them differently in a way zarf could recombine them) or we could push the images back to the registry as they are (though this would require some rebuild logic to get them back into a package). @bburky @jeff-mccoy have some ideas.

Reply:
We can also split image handing into a separate issue since skelly packages do have value alone.

Reply:
By default, only the skelly package is created, users must specify `--include-images` for the images to be included.  Once a determination is made on how we want to handle chunking, we can invert that flag.